src/audio/dai.c

```diff
@@ -540,6 +542,11 @@ static int dai_comp_trigger(struct comp_dev *dev, int cmd)
 	if (ret == COMP_STATUS_STATE_ALREADY_SET)
 		return PPL_STATUS_PATH_STOP;
 
+	/* setup callback */
+	notifier_unregister(dev, dd->chan, NOTIFIER_ID_DMA_COPY);
+	notifier_register(dev, dd->chan, NOTIFIER_ID_DMA_COPY,
+			  dai_dma_cb);
```

Comment:
is this because there's no suitable initialisation location, where a notifier can be registered once?

Reply:
True. Previously these callbacks were registered in config/params, but in multicore scenario, trigger is the first call done on target core, so to register to appropriate notifier (there is 1 per core), I've moved it here.
@tlauda is currently working on moving whole pipeline setup to target core, instead of always building them on master core, which would allow us to put registration back to config/params calls.